mmengine/infer/infer.py

```diff
@@ -0,0 +1,491 @@
+        self,
+        model: Union[ModelType, str],
+        weights: Optional[str] = None,
+        device: Optional[str] = None,
+    ) -> None:
+        self.scope = \
+            DefaultScope.get_current_instance().scope_name  # type: ignore
+        # Load config to cfg
+        cfg: ConfigType
+        if isinstance(model, str):
+            if osp.isfile(model):
+                cfg = Config.fromfile(model)
+            else:
+                # Load config and weights from metafile. If `weights` is
+                # assigned, the weights defined in metafile will be ignored.
+                cfg, _weights = self._load_model_from_metafile(model)
+                if weights is not None:
+                    weights = _weights
+        elif isinstance(model, (Config, ConfigDict)):
+            cfg = copy.deepcopy(model)
+        elif isinstance(model, dict):
+            cfg = copy.deepcopy(ConfigDict(model))
+        else:
+            raise TypeError('config must be a filepath or any ConfigType'
+                            f'object, but got {type(model)}')
+        # Delete the `pretrained` field to prevent model from loading the
+        # the pretrained weights unnecessarily.
+        if cfg.model.get('pretrained') is not None:
+            del cfg.model.pretrained
+
+        if device is None:
+            device = get_device()
+
+        self.cfg = cfg
+        self.model = self._init_model(cfg, weights, device)  # type: ignore
+        self.pipeline = self._init_pipeline(cfg)
+        self.collate_fn = self._init_collate(cfg)
+        self.visualizer = self._init_visualizer(cfg)
+
+    def __call__(
+        self,
+        inputs: InputsType,
+        return_datasamples: bool = False,
+        batch_size: int = 1,
+        **kwargs,
+    ) -> dict:
+        """Call the inferencer.
+
+        Args:
+            inputs (InputsType): Inputs for the inferencer.
+            return_datasamples (bool): Whether to return results as
+                :obj:`BaseDataElement`. Defaults to False.
+            batch_size (int): Batch size. Defaults to 1.
+            **kwargs: Key words arguments passed to :meth:`preprocess`,
+                :meth:`forward`, :meth:`visualize` and :meth:`postprocess`.
+                Each key in kwargs should be in the corresponding set of
+                ``preprocess_kwargs``, ``forward_kwargs``, ``visualize_kwargs``
+                and ``postprocess_kwargs``.
+
+        Returns:
+            dict: Inference and visualization results.
+        """
+        (
+            preprocess_kwargs,
+            forward_kwargs,
+            visualize_kwargs,
+            postprocess_kwargs,
+        ) = self._dispatch_kwargs(**kwargs)
+
+        inputs = self.preprocess(
+            inputs, batch_size=batch_size, **preprocess_kwargs)
+        preds = []
+        for data in track(inputs, description='Inference'):
+            preds.extend(self.forward(data, **forward_kwargs))
+        visualization = self.visualize(
```

Comment:
The visualized image will be saved by VisBackend defined in the config?

Reply:
Maybe, it will also be influenced by the `visualize` and `postprocess` implemented in subclass.